pkg/test-infra/pkg/tidb/recommend.go

```diff
@@ -58,7 +61,20 @@ func (t *TiDBClusterRecommendation) TiDBReplicas(replicas int32) *TiDBClusterRec
 	return t
 }
 
-// tidb-operator do not have sane defaults yet, do a recommendation
+func buildImage(name string) string {
+	var b strings.Builder
+	if fixture.E2eContext.HubAddress != "" {
+		fmt.Fprintf(&b, "%s/", fixture.E2eContext.HubAddress)
+	}
+	b.WriteString(fixture.E2eContext.DockerRepository)
+	b.WriteString("/")
+	b.WriteString(name)
+	b.WriteString(":")
+	b.WriteString(fixture.E2eContext.ImageVersion)
+	return b.String()
+}
+
+// RecommendedTiDBCluster does a recommendation, tidb-operator do not have same defaults yet
```

Comment:
Does this publish the svc for tidb cluster?

Reply:
`*corev1.Service` field in `TiDBClusterRecommendation` will create a nodeport which will public svc for tidb cluster